### DISCUSSION TITLE
Move file from temp directory instead of copying when saving from temp disk is s3

### DISCUSSION THREAD

**Discussion by mkmaker** [livewire/livewire#8994] [3. Feature ideas]
I love the temporary file feature in Livewire and wanted to see how it's handling it.

I see that when I set the temp directory to s3 the file is uploaded directly to s3 from the browser instead of hitting the app server directly.

What I don't like is that when we use method `store()` it will stream and store the file to the designated place. Why not just use move method of s3? 

Because currently it downloads the file in app server and then uploads again to s3 even if we are using s3 as temp directory.

<img width="1138" alt="image" src="https://github.com/user-attachments/assets/066cf2fb-1854-4552-b390-17c4f4ab7b73">

**Comment by pxlrbt** [comment]
Stumbling over this over a year later and wondering the same.
